src/SearchComponents/ExpertSearch.svelte

```diff
@@ -1,57 +1,115 @@
+        let queryString = 'PREFIX knora-api: <http://api.knora.org/ontology/knora-api/v2#>\nPREFIX ' + shortName + ': <http://' + server + '/ontology/' + shortCode + '/' + ontology + '/v2#>\n' + '\nCONSTRUCT {\n';
+        if (enteredString === '') {
+            queryString += '?mainres knora-api:isMainResource true .} WHERE {\n ?mainres a knora-api:Resource .\n?mainres a ' + selectedResource + ' .}';
+        } else {
+            const mainres = enteredString.substring(0, enteredString.indexOf(' '));
+            const lines = enteredString.split('\n');
+            queryString += mainres + ' knora-api:isMainResource true .';
+            for (let line of lines) {
+                if (line.startsWith('FILTER')) {
+                    continue;
+                }
+                if (line.indexOf('knora-api:valueAsString') !== -1 || line.indexOf('knora-api:intValueAsInt') !== -1) {
+                    continue;
+                }
+                const arr = line.split(' ');
+                if (arr.length === 4) {
+                    if (!(arr[2].startsWith('?'))) { // found listnode filter
+                        continue;
+                    }
+                    if (arr[1].indexOf(':') === -1) { // prefix: missing
+                        line = arr[0] + ' ' + shortName + ':' + arr[1] + ' ' + arr[2] + ' ' + arr[3];
+                    }
+                }
+                queryString += '\n' + line;
             }
-            if (o.hasOwnProperty('knora-api:isResourceProperty') && o['knora-api:isResourceProperty']) {
-                for (const r of toReturn){
-                    if (o['knora-api:subjectType']['@id'] === r.id){
-                        r.addProp(new Property(translateLabels(o['rdfs:label']), o['knora-api:objectType']['@id']))
+            queryString += '} WHERE {\n' + mainres + ' a knora-api:Resource .\n' + mainres + ' a ' + selectedResource + ' .';
+            for (let line of lines) {
+                const arr = line.split(' ');
+                if (arr.length === 4) {
+                    if (arr[1].indexOf(':') === -1) { // prefix: missing
+                        line = arr[0] + ' ' + shortName + ':' + arr[1] + ' ' + arr[2] + ' ' + arr[3];
                     }
                 }
+                queryString += '\n' + line;
             }
+            queryString += '}';
         }
-        resources = toReturn;
+        queryString = replaceOntoWithShortName(queryString);
+        const res = await fetch('https://' + server + '/v2/searchextended', {
+            method: 'POST',
+            body: queryString
+
+        })
+        const json = await res.json();
+        console.log(json)
     }
-    getOntology();
-    let selectedResource;
+
+    let resourceProm = getAllResourcesWithLabels();
+
 </script>
-<select bind:value={selectedResource}>
+<select bind:value={selectedResource}
+        on:change={() => {for (const helper of helpers){helper.setDeleted();} noOfProps++; getQuery();}}>
```

Comment:
I would turn this anonymous function into one in the <script> part so it can be tested easily

Reply:
Done